core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/progress/ProgressListenerContext.java

```diff
@@ -0,0 +1,162 @@
+
+    @Override
+    public SdkRequest request() {
+        return sdkRequest;
+    }
+
+    @Override
+    public SdkHttpRequest httpRequest() {
+        return sdkHttpRequest;
+    }
+
+    @Override
+    public ProgressSnapshot uploadProgressSnapshot() {
+        return uploadProgressSnapshot;
+    }
+
+    @Override
+    public ProgressSnapshot downloadProgressSnapshot() {
+        return downloadProgressSnapshot;
+    }
+
+    @Override
+    public SdkHttpResponse httpResponse() {
+        return sdkHttpResponse;
+    }
+
+    @Override
+    public SdkResponse response() {
+        return executionSuccessfulSdkResponse;
+    }
+
+    @Override
+    public String toString() {
+        return ToString.builder("ProgressListenerContext")
+                       .add("sdkRequest", sdkRequest)
+                       .add("uploadProgressSnapshot", uploadProgressSnapshot)
+                       .add("downloadProgressSnapshot", downloadProgressSnapshot)
+                       .add("executionSuccessfulSdkResponse", executionSuccessfulSdkResponse)
+                       .build();
+    }
+
+    public static final class Builder implements CopyableBuilder<Builder, ProgressListenerContext> {
+        private SdkRequest sdkRequest;
+        private SdkHttpRequest sdkHttpRequest;
+        private ProgressSnapshot uploadProgressSnapshot;
+        private ProgressSnapshot downloadProgressSnapshot;
+        private SdkHttpResponse sdkHttpResponse;
+        private SdkResponse executionSuccessfulSdkResponse;
+
+        private Builder() {
+        }
+
+        private Builder(ProgressListenerContext context) {
+            this.sdkRequest = context.sdkRequest;
+            this.sdkHttpRequest = context.sdkHttpRequest;
+            this.uploadProgressSnapshot = context.uploadProgressSnapshot;
+            this.downloadProgressSnapshot = context.downloadProgressSnapshot;
+            this.sdkHttpResponse = context.sdkHttpResponse;
+            this.executionSuccessfulSdkResponse = context.executionSuccessfulSdkResponse;
+        }
+
+        public Builder request(SdkRequest sdkRequest) {
+            this.sdkRequest = sdkRequest;
+            return this;
+        }
+
+        public Builder httpRequest(SdkHttpRequest sdkHttpRequest) {
+            this.sdkHttpRequest = sdkHttpRequest;
+            return this;
+        }
+
+        public Builder uploadProgressSnapshot(ProgressSnapshot uploadProgressSnapshot) {
+            this.uploadProgressSnapshot = uploadProgressSnapshot;
+            return this;
+        }
+
+        public Builder downloadProgressSnapshot(ProgressSnapshot downloadProgressSnapshot) {
+            this.downloadProgressSnapshot = downloadProgressSnapshot;
+            return this;
+        }
+
+        public Builder httpResponse(SdkHttpResponse sdkHttpResponse) {
+            this.sdkHttpResponse = sdkHttpResponse;
+            return this;
+        }
+
+        public Builder response(SdkResponse executionSuccessfulSdkResponse) {
```

Comment:
What is difference between response and httpResponse ?
Can we make naming clear so that we able to distinguish between these two ?

Reply:
The httpResponse is what we get back from the service, instance of the SdkHttpResponse class. The response() method defines the SdkResponse, after the httpResponse is marshalled. We made the naming consistent with how Execution Interceptors have defined them https://github.com/aws/aws-sdk-java-v2/blob/722f187b0df0782568e51f1c402bc0287b504eb4/core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/Context.java#L156